### PR TITLE
Handle pressing all go.bsky.app links in-app w/ resolution

### PR DESCRIPTION
**Pull request by haileyok**
## Why

We want to be able to resolve any short links in the app, even if they don't get resolved while posting. This is a little challenging because we have to reach out to `go.bsky.app` first. I added JSON response support in https://github.com/bluesky-social/social-app/pull/4671, which we can now use to perform resolution when landing on these.

## How

1. First, we click on a link in the app. We already check for if links start with `bsky.app` but we don't check `go.bsky.app` which we add here.
2. We then push to the `StarterPackScreenShort` route
3. The short route attempts to resolve the real URI and then render the starter pack screen with the resolved did and rkey

The nice thing here is this is extensible to other schemes whenever we want to. For example, post short links - which we are less worried about being _very_ short - could use `/p/123456` and we could create a resolve query for those types of links.

## Test Plan

RocketSim_Recording___RNIDE_TEMP_SIM_com apple CoreSimulator SimDeviceType iPhone-15-Pro_6 1_2024-06-27_12 43 20

### CODE REVIEW

File: src/Navigation.tsx
Comment (haileyok) [inline src/Navigation.tsx:334]: A new route that accepts a `code` parameter rather than name and rkey

File: src/lib/strings/url-helpers.ts
Comment (haileyok) [inline src/lib/strings/url-helpers.ts:173]: We don't need to do this for web, so only do it on native. Web can handle the 301 elegantly without weirdness.

File: src/screens/StarterPack/StarterPackLandingScreen.tsx
Comment (haileyok) [inline src/screens/StarterPack/StarterPackLandingScreen.tsx:117]: Just removing this since it is unneeded. No change in functionality.

**Review by gaearon — APPROVED** [review]
have not run, code lgtm (after discussed changes)

**Comment by haileyok** [comment]
Retested signup flow w/ a starter pack, works good. This is a little funky as noted but is mostly a corner case.

**Pull request merged**
